iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergAcidUtil.java

```diff
@@ -64,13 +70,17 @@ private IcebergAcidUtil() {
 
   private static final Types.NestedField PARTITION_HASH_META_COL = Types.NestedField.required(
       MetadataColumns.PARTITION_COLUMN_ID, MetadataColumns.PARTITION_COLUMN_NAME, Types.LongType.get());
+
+  public static final Types.NestedField PARTITION_PROJECTION = Types.NestedField.required(
+      Integer.MAX_VALUE - 6, "partition_projection", Types.StringType.get());
```

Comment:
why `Integer.MAX_VALUE - 6` :) ?

Reply:
Setting a unique integer for partition projection since a unique integer is required for schema.
Particularly Integer.MAX_VALUE - 6 because the current set of predefined list of constants for Iceberg columns have come to Integer.MAX_VALUE - 5.
https://github.com/apache/iceberg/blob/5455d30d9b6e0409d517138e9b80d6831ffb233b/core/src/main/java/org/apache/iceberg/MetadataColumns.java#L56

However, this can be any unique constant as per choice.